modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -80,6 +82,55 @@ func (h *handler) Cleanup(key string, bd *fleet.BundleDeployment) (*fleet.Bundle
 }
 
 func (h *handler) DeployBundle(bd *fleet.BundleDeployment, status fleet.BundleDeploymentStatus) (fleet.BundleDeploymentStatus, error) {
+
+	if bd.Spec.Options.Schedule != "" && status.ScheduledAt == "" {
```

Comment:
What happens if one is unset and the other isn't?

Reply:
There are two ways this can happen:

1. You edit the bundle and remove the Schedule from the bundleSpec. In this scenario, the checks on 86 and 101 will not be met, and the bundle will be deployed immediately.
2. You remove the status.ScheduledAt, this will cause a reconcile of the bundle, and a new ScheduledAt will be applied to bundle, the bundle will be requeued to be applied at the specified time.

Reply:
Alright, that sounds good